src/shared/utils/hooks/useSessionBanners/useSessionBanners.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useRef } from 'react';
+
+import { useBanners } from '~/entities/banner/model';
+import { userModel } from '~/entities/user';
+
+export const useSessionBanners = () => {
+  const { isAuthorized } = userModel.hooks.useAuthorization();
+  const { removeAllBanners } = useBanners();
+
+  const prevIsAuthorized = useRef(isAuthorized);
+
+  useEffect(() => {
+    if (prevIsAuthorized.current !== isAuthorized && !isAuthorized) {
```

Comment:
Wondering why the check for `!isAuthorized` feels like we should also clear notifications when login is successful.

![CleanShot 2024-06-10 at 12 58 54](https://github.com/ChildMindInstitute/mindlogger-web-refactor/assets/2916156/e1301e1d-f70f-4f0b-ae93-d96171ab79aa)

Reply:
The primary reason I opted to do it on logout (in the admin app, and then carried that functionality here), is to give the user confidence that their session is over and their state has been securely cleared (residual banners meant for them aren't bleeding into a now-anonymous session state).

I hear you with this particular banner persistence not making sense post-login success. I think it would make sense to just clear the specific error banner after login is successful, rather than do a blanket `removeAllBanners()` when a session is activated. IMO this same approach would make sense to do for _any_ form submission that at first may generate an error banner, then succeeds before the error banner has timed out. Thus I don't think it belongs in `useSessionBanners`. I'll make this change for the login screen for now.

This hasn't been identified in the admin app, but there may be situations where the same thing makes sense. We might start to see such things reported with more intensive QA.

Reply:
I made that change for both the login & and signup forms (which also exhibited this behaviour, so long as you're quick enough to submit the form successfully after an error is shown).